src/quo/components/profile/expanded_collectible/view.cljs

```diff
@@ -11,38 +11,24 @@
     [react-native.core :as rn]
     [react-native.reanimated :as reanimated]
     [schema.core :as schema]
-    [utils.datetime :as datetime]
     [utils.i18n :as i18n]))
 
-(def timing-options-out 650)
-(def timing-options-in 1000)
-(def first-load-time 500)
-(def cached-load-time 200)
+(def loader-out 650)
+(def image-in 1000)
 (def error-wait-time 800)
 
 (defn on-load-end
-  [{:keys [load-time set-state loader-opacity image-opacity]}]
-  (reanimated/animate loader-opacity 0 timing-options-out)
-  (reanimated/animate image-opacity 1 timing-options-in)
-  (if (> load-time cached-load-time)
-    (js/setTimeout
-     (fn []
-       (set-state (fn [prev-state]
-                    (assoc prev-state :image-loaded? true))))
-     first-load-time)
-    (set-state (fn [prev-state]
-                 (assoc prev-state :image-loaded? true)))))
+  [{:keys [loader-opacity image-opacity]}]
+  (reanimated/animate loader-opacity 0 loader-out)
+  (reanimated/animate image-opacity 1 image-in))
 
 (defn on-load-error
-  [set-state]
-  (js/setTimeout (fn []
-                   (set-state (fn [prev-state] (assoc prev-state :image-error? true))))
-                 error-wait-time))
+  [set-error]
+  (js/setTimeout set-error error-wait-time))
```

Comment:
Simplified the calculations for the animation, it's still looks good but the implementation is a bit simpler